maven-plugin-api/pom.xml

```diff
@@ -54,6 +54,11 @@ under the License.
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-classworlds</artifactId>
     </dependency>
+    <dependency>
```

Comment:
can we not use it and just test with a custom Log impl (like an in memory one or reusing default/logger one)? I don't see what it brings but I see how the test can pass with a broken impl ;)

Reply:
That was just a simple unit test, as far as possible I didn't wanted to interfere/pollute the console outputs.
If it is not a common practice to mock then I can add a stub/custom impl (with control on log level) for the same purpose.

Reply:
> with a custom Log impl

you probably meant with a org.codehaus.plexus.logging.Logger implementation ; applying to tests in DefaultLogTest.

@rmannibucau If I provide a stub/custom impl what is the difference with the mock ; I mean in regard of the remark _"... what it brings but I see how the test can pass with a broken impl"_

Reply:
Well for the last part mockito does way more than a custom impl and half of it is not desired I guess for such tests (mainly cause of maintenance).
On the logger impl I meant DefaultLog impl whatever is under (if it moves to something else than plexus and uses a slf4j particular binding it would be the kind of same comment).

Reply:
I will remove mockito & change to custom/stub impl and change the tests accordingly

Reply:
mockito removed and custom impl provided where needed.